src/software/sensor_fusion/filter/possession_filter.h

```diff
@@ -0,0 +1,35 @@
+#pragma once
+
+#include <vector>
+
+#include "shared/constants.h"
+#include "software/world/ball.h"
+#include "software/world/robot_state.h"
+#include "software/world/team.h"
+#include "software/world/timestamped_possession_state.h"
+
+/**
+ * Compiles the list of robots that have possession of the ball
+ *
+ * @param friendly_robots_with_breakbeam_triggered The friendly robots that have their
+ * breakbeam triggered
+ * @param friendly_team The friendly_team of robots
+ * @param enemy_team The enemy_team of robots
+ * @param ball The ball
+ *
+ * @return the list of robots that possession of the ball
+ */
+std::vector<RobotIdWithTeamSide> getRobotsWithPossession(
+    std::vector<RobotId> friendly_robots_with_breakbeam_triggered, Team friendly_team,
```

Comment:
Would we need `friendly_robots_with_breakbeam_triggered` if the Teams had `ballInMouth`? I would expect setting `ballInMouth` to use breakbeam if available rather than providing it here.

Reply:
the ballInMouth PR got closed https://github.com/UBC-Thunderbots/Software/pull/1477 so Team has not knowledge of breakbeam. It needs to be passed in separately if we want to keep it in the SensorFusion layer (and not give AI info of ballInMouth)

Reply:
I thought we were going to add `ballInMouth` to the `Robot` class, just not the `RobotState` class? In that case the Teams would have access to that information.